### DISCUSSION TITLE
Region virtualisation: rapid removal and creation of large numbers of regions appears to be glitchy since 7.8.0

### DISCUSSION THREAD

**Discussion by rewbs** [katspaugh/wavesurfer.js#3841] [Q&A] [answered]
Hi all,

I have a UI that uses regions to represent beat positions in a track. The region positions are calculated based on the BPM and beat offset. The user can change the BPM and beat offset with a slider, which updates the beat markers in real time.

The way I'm doing this is quite naive, but worked well up until 7.8.0. When the BPM or offset changes,  I simply delete all the "beat" regions like `regionsPlugin.getRegions().filter(r => r.id.includes('beat')).forEach(r => r.remove())` (I can't simply clear all regions because there are non-beat regions I need to keep) , and then re-invoke regionsPlugin.addRegion() for each updated beat position.

Here it is in action in 7.7.15:

https://github.com/user-attachments/assets/30c7b0ad-31f1-4905-b77a-e11a36c638fb

But as of 7.8.0, there is a lag and a refresh issue (only change here is the WaveSurfer version, no change in my code):

https://github.com/user-attachments/assets/b9263589-408e-45ce-b37b-e2094233adc5

Does this feel like a regression in Wavesurfer or is my approach brain-dead? I notice 7.8.0 introduced some virtualisation for regions, which I assume is related.

For further context I'm using @wavesurfer/react 1.0.7  and Chrome on MacOS.

If there's consensus this is a bug or unexpected (as opposed to me being an idiot), I'm happy to work on a simple recreate case.

Cheers,
Robin

**Comment by katspaugh** [comment, marked as answer]
Thanks Robin, it's definitely a regression! I'll look into it.

**Comment by jomin398** [comment]
@rozap can you share your code?  
Visualizing the bpm is overlaps with what I'm trying to make.
I'm trying to cloning to look like pioneer's DJ machine.  
![BrowserPreview_tmp_02.png](https://github.com/user-attachments/assets/2ff23c85-15e7-4a29-9ffb-8657cee3a387)

May be I'm going to make a plugin.

**Comment by katspaugh** [comment]
@rewbs is this still the case with the latest version (7.8.13)? There were a couple fixes for region removal.
